pkg/controllers/etcd.go

```diff
@@ -87,6 +113,17 @@ func (s *EtcdService) Run(ctx context.Context, ready chan<- struct{}, stopped ch
 	klog.Info("etcd is ready!")
 	close(ready)
 
+	// Once we're ready and running as a service, tell systemd that we're ready.
+	if runningAsSvc {
+		if supported, err := daemon.SdNotify(false, daemon.SdNotifyReady); err != nil {
```

Comment:
I think this needs to be in the `microshift-etcd` binary - `microshift` already has this and we don't want to publish this just after etcd is up, but after every component is up and running.

Let me rephrase my idea:
- `microshift` executes `systemd-run ... microshift-etcd run` with blocking `Run()`
- `microshift-etcd` can:
  - exit with error code (which systemd-run should propagate)
  - timeout - didn't sent READY ntfy in N seconds (would 30 be a sane default?), systemd-run kills it and propagates error
  - send READY, continue running in background, systemd-run exits with success (0)
- `Run()` function returns, `microshift` inspects error code, if success, then process with testing etcd endpoint, if failure then exit

All of this is based on a assumption that systemd supports and honors `SdNotifyReady` for transient units. Because we're not monitoring unit's health, it could be that we fire and forget the systemd-run but are left without etcd, right?

Reply:
Thanks for rephrasing, that makes much more sense!